### ISSUE TITLE
memcached error - no activities are being shown

### ISSUE THREAD

**Issue by beedaddy** [owncloud/activity#476]
### Steps to reproduce
1. Just go to the activities app

### Expected behaviour
Some activities should be shown.

### Actual behaviour
No activities are being shown. Instead, I get this error message in the log:
`Error 33 interacting with memcached : A BAD KEY WAS PROVIDED\\\/CHARACTERS OUT OF RANGE\`

### Server configuration
**Operating system**: Debian 8

**Web server:** Apache 2.4

**Database:** PostgreSQL 9.4

**PHP version:** 5.6

**ownCloud version:** 9.0.0

**Updated from an older ownCloud or fresh install:** Update from 8.2.3 

**Where did you install ownCloud from:** tar.bz2 from owncloud.org

**Signing status (ownCloud 9.0 and above):**
```
Results
=======
- core
	- INVALID_HASH
		- .htaccess

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [.htaccess] => Array
                        (
                            [expected] => f40e7ca19cd76ff922ccd164e38f4d73806f66170cafa202fb470be225c813b4b242f4321da8a01bef1329e300acf3d137dbcbff058c501f166a2619031fc651
                            [current] => 7bae810b6f5ca98d062ac9985cac7fb2dc8e22d981ec41e8327e53a5ee7f33017b128bbb4ceefbe665dbd35f7a1ee3a0926c4278aeeeb6d7b4325197be1ed9a9
                        )

                )

        )

)
```

**List of activated apps:**
```
Enabled:
  - activity: 2.2.1
  - announcementcenter: 1.1.0
  - calendar: 1.0
  - comments: 0.2
  - contacts: true
  - dav: 0.1.5
  - documents: 0.12.0
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - news: true
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - tasks: true
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - encryption
  - external
  - user_external
  - user_ldap
```

**The content of config/config.php:**
```
{
    "system": {
        "instanceid": "51adb70420b3a",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "\/var\/owncloud-data",
        "dbtype": "pgsql",
        "version": "9.0.0.19",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance": false,
        "forcessl": true,
        "loglevel": 2,
        "log_rotate_size": 2097152,
        "theme": "",
        "maxZipInputSize": 1610612736,
        "allowZipDownload": true,
        "trusted_domains": [
            "owncloud.brodbeck-online.de",
            "brodbeck-online.de"
        ],
        "secret": "***REMOVED SENSITIVE VALUE***",
        "appstoreenabled": true,
        "appstore.experimental.enabled": true,
        "filelocking.enabled": "true",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": 6379,
            "timeout": 0,
            "dbindex": 0
        },
        "memcached_servers": [
            [
                "localhost",
                11211
            ]
        ],
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Memcached",
        "trashbin_retention_obligation": "auto"
    }
}
```

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

### Client configuration
**Browser:** Firefox 44.0.2

**Operating system:** Arch Linux

**Comment by nickvergessen** [comment]
Please report in core, since that is not an error of the activity app, thanks:
https://github.com/owncloud/core/issues/new

**Comment by nickvergessen** [comment]
Or can you please show all statements of your owncloud.log file?

**Comment by beedaddy** [comment]
Sure, here it is:
`{"reqId":"xxx","remoteAddr":"xxx.xx.xxx.xx","app":"index","message":"Exception: {\"Exception\":\"Exception\",\"Message\":\"Error 33 interacting with memcached : A BAD KEY WAS PROVIDED\\\/CHARACTERS OUT OF RANGE\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/memcache\\\/memcached.php(80): OC\\\\Memcache\\\\Memcached->verifyReturnCode()\\n#1 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/route\\\/cachingrouter.php(57): OC\\\\Memcache\\\\Memcached->set('owncloud.brodbe...', '\\\/apps\\\/files\\\/?di...', 3600)\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/urlgenerator.php(66): OC\\\\Route\\\\CachingRouter->generate('files.view.inde...', Array)\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/urlgenerator.php(79): OC\\\\URLGenerator->linkToRoute('files.view.inde...', Array)\\n#4 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/formatter\\\/fileformatter.php(91): OC\\\\URLGenerator->linkToRouteAbsolute('files.view.inde...', Array)\\n#5 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/parameter\\\/parameter.php(82): OCA\\\\Activity\\\\Formatter\\\\FileFormatter->format(Object(OC\\\\Activity\\\\Event), Array)\\n#6 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/parameter\\\/collection.php(88): OCA\\\\Activity\\\\Parameter\\\\Parameter->format()\\n#7 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/datahelper.php(87): OCA\\\\Activity\\\\Parameter\\\\Collection->format()\\n#8 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/datahelper.php(129): OCA\\\\Activity\\\\DataHelper->translation('files', 'created_self', Array)\\n#9 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/grouphelper.php(207): OCA\\\\Activity\\\\DataHelper->formatStrings(Array, 'subject')\\n#10 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/data.php(268): OCA\\\\Activity\\\\GroupHelper->getActivities()\\n#11 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/controller\\\/ocsendpoint.php(209): OCA\\\\Activity\\\\Data->get(Object(OCA\\\\Activity\\\\GroupHelper), Object(OCA\\\\Activity\\\\UserSettings), 'martin', 0, 50, 'desc', 'all', '', 0)\\n#12 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/controller\\\/ocsendpoint.php(180): OCA\\\\Activity\\\\Controller\\\\OCSEndPoint->get(Array)\\n#13 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/controller\\\/endpoint.php(68): OCA\\\\Activity\\\\Controller\\\\OCSEndPoint->getFilter(Array)\\n#14 [internal function]: OCA\\\\Activity\\\\Controller\\\\EndPoint->getFilter('all')\\n#15 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/appframework\\\/http\\\/dispatcher.php(159): call_user_func_array(Array, Array)\\n#16 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/appframework\\\/http\\\/dispatcher.php(89): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OCA\\\\Activity\\\\Controller\\\\EndPoint), 'getFilter')\\n#17 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/appframework\\\/app.php(110): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OCA\\\\Activity\\\\Controller\\\\EndPoint), 'getFilter')\\n#18 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/appframework\\\/routing\\\/routeactionhandler.php(45): OC\\\\AppFramework\\\\App::main('EndPointControl...', 'getFilter', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#19 [internal function]: OC\\\\AppFramework\\\\routing\\\\RouteActionHandler->__invoke(Array)\\n#20 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/route\\\/router.php(273): call_user_func(Object(OC\\\\AppFramework\\\\routing\\\\RouteActionHandler), Array)\\n#21 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(873): OC\\\\Route\\\\Router->match('\\\/apps\\\/activity\\\/...')\\n#22 \\\/var\\\/www\\\/owncloud\\\/index.php(39): OC::handleRequest()\\n#23 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/memcache\\\/memcached.php\",\"Line\":174}","level":3,"time":"2016-03-10T08:56:13+00:00"}
`

**Comment by beedaddy** [comment]
Sorry it might just has been a config problem. I removed the memcached stuff from the config file (I accidentially had both memcached and redis in my config file) and it now seems to work.

**Comment by mmccarn** [comment]
I'm seeing the same behavior - also related to memcached.  
With memcache.local or memcache.distributed set, the Activity app shows a spinning progress indicator forever.

With both of these disabled, Activity works fine (but I get a 'no memory cache' warning).

**Operating system**: Ubuntu 14.04
**Web server**: Apache 2.4.7
**Database**: 5.5.47-0ubuntu0.14.04.1 (Ubuntu)
**PHP version**: PHP 5.5.9-1ubuntu4.14
**ownCloud **version: 9.0.0.0.19
**Updated from an older ownCloud or fresh install**: Update from 8.2.2
**Where did you install ownCloud from**: apt-get http://download.owncloud.org/download/repositories/stable/xUbuntu_14.04/

I use memcached instead of APCu or REDIS as it is the only cache solution supported by owncloud that is also supported by default in Ubuntu 14.04.

The required versions of APCu or REDIS must either be installed from source or from the backports repository - but https://help.ubuntu.com/community/UbuntuBackports says:

> Unlike the packages released with Ubuntu, Backports do not come with any security support guarantee.

**Comment by nickvergessen** [comment]
Well it will be fixed in 9.0.1 which is released next week (most likely)

See https://github.com/owncloud/core/pull/23214 and try to apply it yourself.

**Comment by mmccarn** [comment]
I can confirm that the very small change from the commit you mention solved the problem for me.

Thanks.
